checkers/javascript/location_assign.test.js

```diff
@@ -0,0 +1,9 @@
+// Match
+// <expect-error>
+location.assign(document.getElementById("input").value);
+// <expect-error>
+location.assign(new URLSearchParams(window.location.search).get("redirect"));
+// <expect-error>
+window.location.assign(document.getElementById("input").value);
+// <expect-error>
+window.location.assign(new URLSearchParams(window.location.search).get("redirect"));
```

Comment:
Also add cases where no error should be raised.

Test for cases like this as well:
```javascript
location.assign("https://example.com" + new URLSearchParams(window.location.search).get("redirect"));
```